numpy/core/setup.py

```diff
@@ -605,6 +606,12 @@ def generate_api(ext, build_dir):
     config.add_include_dirs(join('src', 'umath'))
     config.add_include_dirs(join('src', 'npysort'))
 
+    # Need to add location of npy_math_internal.h, generated from src/npymath/npy_math_internal.h.src
+    npymath_build_dir = join('build', 'src.%s-%s.%s' % (sysconfig.get_platform(), sys.version_info[0], sys.version_info[1]),
```

Comment:
@juliantaylor,  could you please describe situation where it can break?

The way I see it is.. if Numpy was built in one env, say having all the HAVE_ macros, and another environment says that there is no some or all of the corresponding functions, wouldn't npymath.a library be incompatible as well? The vice versa situation is actually safe though a little bit non-optimal for static library case but might be more optimal for the inline case since external module might have it's own configure script which defines HAVE_ macros which did not exist for Numpy build.

In any way, we might want to make it opt-in for external modules with this `NPY_INLINE_MATH` macro which is enough for many distributions while safe enough for personal builds